### DISCUSSION TITLE
Send flag to Unreal through API

### DISCUSSION THREAD

**Discussion by Sabrin0** [microsoft/AirSim#4208] [Support Q&A] [answered]
Is there a way to send a signal to unreal via Python API? I would like to execute BP functions only under certain conditions, for instance whenever the  lidar detects obstacles close to the multirotor

**Comment by rajat2004** [comment, marked as answer]
The [`simRunConsoleCommand`](https://microsoft.github.io/AirSim/api_docs/html/#airsim.client.VehicleClient.simRunConsoleCommand) might work for you to run execute Unreal commands from the Python API

**Reply by Sabrin0** [reply]
Yes, I did so. Thank you for your answer

**Reply by Sabrin0** [reply]
Hi @rajat2004, is it also possible to send data from python API to Unreal BP?

**Reply by rajat2004** [reply]
If the above API isn't sufficient, you can always write your own APIs, see the docs https://microsoft.github.io/AirSim/adding_new_apis/ on how to do this, alongwith some examples. Feel free to reach out here if running into some problems
